qa/SidechainTestFramework/account/ac_use_smart_contract.py

```diff
@@ -0,0 +1,220 @@
+import json
+from typing import Tuple
+from eth_abi import encode_abi, decode_abi
+from SidechainTestFramework.account.ac_smart_contract_compile import prepare_resources, get_cwd
+import os
+from dataclasses import dataclass
+from SidechainTestFramework.account.mk_contract_address import mk_contract_address
+
+
+@dataclass
+class ContractFunction:
+    name: str
+    inputs: []
+    outputs: []
+    sigHash: str
+    isConstructor: bool
+
+    def encode(self, *inputs):
+        if not self.isConstructor:
+            encoded_string = self.sigHash + encode_abi(self.inputs, inputs).hex()
+        else:
+            encoded_string = encode_abi(self.inputs, inputs).hex()
+        return encoded_string
+
+    def decode(self, output):
+        return decode_abi(self.outputs, output)
+
+    def __str__(self):
+        return f'{self.sigHash} :: {self.name}({",".join(self.inputs)}) -> ({",".join(self.outputs)})'
+
+
+class SmartContract:
```

Comment:
let's add the simple description and docs for the public methods of this class: what are inputs, outputs, etc.